config/fi_check_package.m4

```diff
@@ -130,7 +130,7 @@ AC_DEFUN([_FI_CHECK_PACKAGE_LIB], [
                           unset fi_Lib])])])])
 
     AS_IF([test "$fi_check_package_lib_happy" = "yes"],
-          [$1_LIBS="-l$2 $4"
+          [$1_LIBS="-l$2 $4 $$1_LIBS"
```

Comment:
When do you call `FI_CHECK_PACKAGE` a second time with the same prefix?

It looks like the verbs provider is doing this (two calls to `FI_CHECK_PACKAGE` for `verbs_ibverbs`), but that doesn't look like the Right Thing -- the 2nd call might be able to simply be a call to `AC_CHECK_HEADERS([infiniband/verbs_exp.h])`...?